### PR TITLE
Add week number support

### PR DESCRIPTION
**Pull request by judikdavid**
Adds week number in front of every week line in the first column.
It uses a new withWeekNumbers prop on every component, that toggles the
feature.
It supports both horizontal and vertical datepickers.

### CODE REVIEW

**Comment by majapw** [comment]
Can we see some screenshots?

**Comment by judikdavid** [comment]
Of course, see below:
screen shot 2016-09-12 at 22 11 19
screen shot 2016-09-12 at 22 11 38

File: css/CalendarMonth.scss [hunk 1]
```diff
@@ -36,6 +36,20 @@
...
 
+.CalendarMonth__weeknumber {
+  padding: 0;
+  box-sizing: border-box;
+  color: $react-dates-color-gray;
+  cursor: default;
+  width: 39px;
+  height: 38px;
+
+  user-select: none;
```
Comment (majapw) [inline css/CalendarMonth.scss]: Can you test that this `user-select` is still necessary given https://github.com/airbnb/react-dates/pull/74 and follow up commit?

File: src/components/CalendarMonth.jsx [hunk 2]
```diff
@@ -53,11 +55,26 @@ export function getModifiersForDay(modifiers, day) {
   return day ? Object.keys(modifiers).filter(key => modifiers[key](day)) : [];
 }
 
+export function getWeekNumberForWeek(week) {
```
Comment (majapw) [inline src/components/CalendarMonth.jsx]: Can we pull this into its own `utils` file a la `getsCalendarMonthWeeks`?

File: src/components/CalendarMonth.jsx [hunk 3]
```diff
@@ -53,11 +55,26 @@ export function getModifiersForDay(modifiers, day) {
...
+  if (week && Array.isArray(week)) {
+    const validDays = week.filter((d) => (d));
+    if (validDays[0] && moment.isMoment(validDays[0])) {
+      const firstDayOfWeek = validDays[0].clone().startOf('week');
+
+      return firstDayOfWeek.week();
+    }
+    return null;
```
Comment (majapw) [inline src/components/CalendarMonth.jsx]: probs we just need the return at the end and not this one, right?

File: src/components/CalendarMonth.jsx [hunk 4]
```diff
@@ -53,11 +55,26 @@ export function getModifiersForDay(modifiers, day) {
   return day ? Object.keys(modifiers).filter(key => modifiers[key](day)) : [];
 }
 
+export function getWeekNumberForWeek(week) {
+  if (week && Array.isArray(week)) {
```
Comment (majapw) [inline src/components/CalendarMonth.jsx]: You seem to be doing a whole lot of defensive checks here that I'm not sure are necessary. `week` is always going to be of a v. specific form and I also don't know that you need to clone the moment objects or call startOf('week') in this way.

File: src/components/CalendarMonthGrid.jsx [hunk 5]
```diff
@@ -60,6 +64,19 @@ const defaultProps = {
   monthFormat: 'MMMM YYYY', // english locale
 };
 
+function getCalendarMonthGridWidthStyle(orientation, withWeekNumbers) {
```
Comment (majapw) [inline src/components/CalendarMonthGrid.jsx]: :| I really want to work on the with-styles change more now cos this sucks that we have to do this.

File: css/CalendarMonth.scss (hunk 1 above)
Comment (judikdavid) [inline css/CalendarMonth.scss]: Yes, you're right, it's not necessary, deleting it in a next commit.

File: src/components/CalendarMonth.jsx (hunk 2 above)
Comment (judikdavid) [inline src/components/CalendarMonth.jsx]: Fixing it in a next commit.

File: src/components/CalendarMonth.jsx (hunk 4 above)
Comment (judikdavid) [inline src/components/CalendarMonth.jsx]: Yes, I overcomplicated things, fixing it.

File: src/components/CalendarMonth.jsx (hunk 3 above)
Comment (judikdavid) [inline src/components/CalendarMonth.jsx]: Right.

File: src/components/CalendarMonthGrid.jsx (hunk 5 above)
Comment (judikdavid) [inline src/components/CalendarMonthGrid.jsx]: Yes, I also think that this is not the best solution, but at least the width value is not duplicated in css and js this way.

File: css/CalendarMonth.scss [hunk 6]
```diff
@@ -36,6 +36,15 @@
   margin-bottom: 2px;
 }
 
+.CalendarMonth__weeknumber {
+  padding: 0;
+  box-sizing: border-box;
+  color: $react-dates-color-gray;
+  cursor: default;
+  width: 39px;
```
Comment (majapw) [inline css/CalendarMonth.scss]: Maybe we should pull these values out into a variable so we can share between the `.CalendarMonth__day` class and this one

File: src/components/CalendarMonthGrid.jsx (hunk 5 above)
Comment (majapw) [inline src/components/CalendarMonthGrid.jsx]: Yeah that totally makes sense! :) Can we pull this out as well into a util method? I'm very much about small utility modules wherever possible. It makes testing a dream.

**Comment by majapw** [comment]
Whoa this new github review flow is fantastic.

**Comment by jontelm** [comment]
Any updates on this one? @judikdavid I want this stuff :)

**Comment by majapw** [comment]
I'm going to close this as this still had some issues with animations at the time of last commit and now will be in conflict hell if we try to reopen it. We can re-entertain this issue in the future if desired.

**Comment by grenhall** [comment]
I'd say this feature is highly requested from everyone in the Nordic region.

**Pull request closed without merging**